assets/prototype/domain/shared/predicates/datetimes/validInfiniteReglimit.ts

```diff
@@ -0,0 +1,27 @@
+/**
+ * External dependencies
+ */
+import { is } from 'ramda';
+
+/**
+ * Internal dependencies
+ */
+import { Datetime } from '../../../eventEditor/data/types';
+
+/**
+ * @param {Object} date event date object
+ * @return {boolean} true if regLimit property is valid
+ */
+const validReglimit = (date: Datetime) => {
+	return is(String, date.regLimit) || is(Number, date.regLimit);
+};
+
+/**
+ * @param {Object} date event object
+ * @return {boolean} true if regLimit property is valid and unlimited
+ */
+const validInfiniteReglimit = (date: Datetime) => {
+	return validReglimit(date) && (date.regLimit === 'INF' || date.regLimit === Infinity);
```

Comment:
If you see the GQL schema, `date.capacity` is always an integer where `-1` represents infinite capacity.

Reply:
Updated: https://github.com/eventespresso/event-espresso-core/pull/2087/commits/b6fa218c0c115b15d7f4dc043f82835a6dd97ccc